RTNetwork.cs

```diff
@@ -204,38 +243,48 @@ internal int Receive(ref byte[] receivebuffer, int offset, int bufferSize, bool
                 if (receiveList.Count == 0)
                 {
                     receivebuffer = null;
-                    return 0;
+                    mErrorString = "No clients to receive from.";
+                    return new Response(Response.ResponseType.error, 0);
                 }
 
                 Socket.Select(receiveList, null, errorList, timeout);
 
+
                 if (mTCPClient != null && errorList.Contains(mTCPClient.Client))
                 {
                     // Error from TCP socket
                     mErrorString = "Error reading from TCP socket";
+                    return new Response(Response.ResponseType.error, 0);
                 }
                 else if (mTCPClient != null && receiveList.Contains(mTCPClient.Client))
                 {
                     // Receive data from TCP socket
-                    return mTCPClient.Client.Receive(receivebuffer, offset, header ? RTProtocol.Constants.PACKET_HEADER_SIZE : bufferSize, SocketFlags.None);
+                    int received = mTCPClient.Client.Receive(receivebuffer, offset, header ? RTProtocol.Constants.PACKET_HEADER_SIZE : bufferSize, SocketFlags.None);
+                    return new Response((received == 0) ? Response.ResponseType.disconnect : Response.ResponseType.success, received);

```

Comment:
Is this correct? That if we receive no bytes we for sure disconnected? `Client.Receive(...)` has an overload with error code as out-parameter. Maybe that should be used instead.

Reply:
This is correct according to the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.receive?view=net-5.0), 0 is returned if connection is shutdown by the remote host.